tapchannelmsg/wire_msgs.go

```diff
@@ -0,0 +1,354 @@
+// during the asset funding process.
+type AssetFundingMsg interface {
+	lnwire.Message
+
+	// PendingChanID is the temporary channel ID that was assigned to the
+	// channel.
+	PendingChanID() funding.PendingChanID
+
+	// Amt returns the amount of the asset that is committed to the channel.
+	Amt() fn.Option[uint64]
+}
+
+// TxAssetInputProof is sent by the initiator of a channel funding request to
+// prove to the upcoming responder that they are the owner of an asset input.
+//
+// TODO(roasbeef): once we have fixed the asset ownership proof to sign a
+// challenge value, we can use temp chan ID as the challenge.
+type TxAssetInputProof struct {
+	// TempChanID is the temporary channel ID that was assigned to the
+	// channel.
+	TempChanID tlv.RecordT[tlv.TlvType0, funding.PendingChanID]
+
+	// AssetID is the ID of the asset that this output is associated with.
+	AssetID tlv.RecordT[tlv.TlvType1, asset.ID]
+
+	// Amount is the amount of the asset that this output represents.
+	Amount tlv.RecordT[tlv.TlvType2, uint64]
+
+	// Proof is the last transition proof that proves this output was
+	// committed to in the Bitcoin transaction that anchors this asset
+	// output.
+	Proof tlv.RecordT[tlv.TlvType3, proof.Proof]
+}
+
+// NewTxAssetInputProof creates a new TxAssetInputProof message.
+func NewTxAssetInputProof(pid funding.PendingChanID,
+	p proof.Proof) *TxAssetInputProof {
+
+	return &TxAssetInputProof{
+		TempChanID: tlv.NewPrimitiveRecord[tlv.TlvType0](pid),
+		AssetID:    tlv.NewRecordT[tlv.TlvType1](p.Asset.ID()),
+		Amount: tlv.NewPrimitiveRecord[tlv.TlvType2](
+			p.Asset.Amount,
+		),
+		Proof: tlv.NewRecordT[tlv.TlvType3](p),
+	}
+}
+
+// MsgType returns the type of the message.
+func (t *TxAssetInputProof) MsgType() lnwire.MessageType {
+	return TxAssetInputProofType
+}
+
+// Decode reads the bytes stream and converts it to the object.
+func (t *TxAssetInputProof) Decode(r io.Reader, _ uint32) error {
+	stream, err := tlv.NewStream(
+		t.TempChanID.Record(),
+		t.AssetID.Record(),
+		t.Amount.Record(),
+		t.Proof.Record(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return stream.Decode(r)
+}
+
+// Encode converts object to the bytes stream and write it into the
+// write buffer.
+func (t *TxAssetInputProof) Encode(w *bytes.Buffer, _ uint32) error {
+	stream, err := tlv.NewStream(
+		t.TempChanID.Record(),
+		t.AssetID.Record(),
+		t.Amount.Record(),
+		t.Proof.Record(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return stream.Encode(w)
+}
+
+// PendingChanID returns the temporary channel ID that was assigned to the
+// channel.
+func (t *TxAssetInputProof) PendingChanID() funding.PendingChanID {
+	return t.TempChanID.Val
+}
+
+// FundingAssetID returns the asset ID of the underlying asset.
+func (t *TxAssetInputProof) FundingAssetID() fn.Option[asset.ID] {
+	return fn.Some(t.AssetID.Val)
+}
+
+// Amt returns the amount of the asset that this output represents.
+func (t *TxAssetInputProof) Amt() fn.Option[uint64] {
```

Comment:
I think it's because some messages don't actually have an amount, but each message needs to implement this to satisfy the interface. IMO the interface itself is a bit awkward, at least since the addition of the `AssetFundingAck`. But changing this would lead to quite a bit of refactor, so I think we should do that at a later point in time.